code/modules/reagents/chemistry/machinery/chem_dispenser.dm

```diff
@@ -65,6 +72,13 @@
 	if(energy != oldenergy)
 		use_power(2500)
 
+/obj/machinery/chem_dispenser/emag_act(mob/user)
+	if(emagged || !emagged_reagents)
+		user << "<span class='warning'>\The [src] has no functional safeties to emag.</span>"
+		return
+	user << "<span class='notice'>You short out \the [src]'s safeties.</span>"
+	dispensable_reagents |= emagged_reagents//add the emagged reagents to the dispensable ones
```

Comment:
Maaaaybe because you did not set the `emagged` var, and the `emagged_reagents` var is a list and cannot be checked properly in the manner you provided. If you set the `emagged` var, you do not need to check `emagged_reagents` anyway.